andela/eventx
# #108727880 Scan Tickets

#### What does this PR do?
Users that have been appointed as Gate Keepers can scan tickets
Non-authorized users will be redirected to the home page
Invalid tickets will return a notification that `Ticket not valid`
Valid tickets will be displayed along with the status `Valid` or `Used`
All tickets will include the owner
Scanned tickets, in addition, include the Gate Keeper who scanned it and when it was scanned
#### Description of Task to be completed?
Add migration to include `is_used`, `time_used`, and `scanned_by`
Use `cancancan` gem to manage permissions
Write controller and feature tests
#### How should this be manually tested?
On the user dashboard, click on scan tickets
Type in an invalid ticket number and see the response
Type in a valid ticket number and see the response
Click on scan for a valid ticket
#### Screenshots
![image](https://cloud.githubusercontent.com/assets/5628767/15147566/32ec4ee4-16b9-11e6-8be9-4230a60463ed.png)
